cmake/HPX_SetupHIP.cmake

```diff
@@ -0,0 +1,50 @@
+# Copyright (c)      2020 ETH Zurich
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+# No need for the user to specify the option explicitly
+if("${CMAKE_CXX_COMPILER}" MATCHES "hipcc$")
+  include(HPX_Option)
+  hpx_option(
+    HPX_WITH_HIP BOOL "Enable compilation with HIPCC (default: ON)" ON ADVANCED
+  )
+else()
+  hpx_option(
+    HPX_WITH_HIP BOOL "Enable compilation with HIPCC (default: OFF)" OFF
+    ADVANCED
+  )
+endif()
+
+if(HPX_WITH_HIP)
+
+  if(HPX_WITH_CUDA)
+    hpx_error(
+      "Both HPX_WITH_CUDA and HPX_WITH_HIP are ON. Please choose one of \
+    them for HPX to work properly"
+    )
+  endif(HPX_WITH_CUDA)
+
+  # Needed on rostam
+  list(APPEND CMAKE_PREFIX_PATH $ENV{HIP_PATH}/lib/cmake/hip)
+  list(APPEND CMAKE_PREFIX_PATH $ENV{DEVICE_LIB_PATH}/cmake/AMDDeviceLibs)
+  list(APPEND CMAKE_PREFIX_PATH $ENV{DEVICE_LIB_PATH}/cmake/amd_comgr)
+  list(APPEND CMAKE_PREFIX_PATH $ENV{DEVICE_LIB_PATH}/cmake/hsa-runtime64)
```

Comment:
hipblas have a `find_dependency(HIP)` which fails if those are not set, I'm not sure why it doesn't look for environmental variables, but it's not needed on ault, so I'll try to investigate

Reply:
Yes hip is found without the need of those variables but hipblas is not because the `find_dependency(HIP)` inside `hipblas-config` fail to find the config files associated with AMDDeviceLibs, amd_comgr, hip and hsa-runtime64